### PR TITLE
refactor(Scripts/RuinsOfAhnQiraj): Clean up Ayamiss script

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp [hunk 1]
```diff
@@ -379,8 +337,7 @@ class spell_ayamiss_swarmer_start_loop : public SpellScript
 
     void HandleScript(SpellEffIndex /*effIndex*/)
     {
-        uint32 loopSpells[3] = { SPELL_HIVEZARA_SWARMER_LOOP_1, SPELL_HIVEZARA_SWARMER_LOOP_2, SPELL_HIVEZARA_SWARMER_LOOP_3 };
-        GetCaster()->CastSpell((Unit*)nullptr, Acore::Containers::SelectRandomContainerElement(loopSpells));
+        GetCaster()->GetAI()->DoCastAOE(RAND(SPELL_HZ_SWARMER_LOOP_1, SPELL_HZ_SWARMER_LOOP_2, SPELL_HZ_SWARMER_LOOP_3));
```
Comment (sudlud) [inline src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp:340]: Can GetAI() return nullptr?

File: src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp [hunk 2]
```diff
@@ -253,29 +234,6 @@ struct npc_hive_zara_larva : public ScriptedAI
...
-        ScriptedAI::AttackStart(victim);
-    }
-
-    void UpdateAI(uint32 diff) override
-    {
-        if (_instance->GetBossState(DATA_AYAMISS) == IN_PROGRESS)
-            return;
-
-        ScriptedAI::UpdateAI(diff);
-    }
```
Comment (sudlud) [inline src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp:278]: What's the reason for dropping all of these?
Or is this replaced by setting it to "Passive" initially

File: src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp (hunk 1 above)
Comment (Nyeriah) [reply src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp:340]: Can very early during object creation. Will never be the case here.

File: src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp (hunk 2 above)
Comment (Nyeriah) [reply src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_ayamiss.cpp:278]: Yes.

**Review by sudlud — APPROVED** [review]

**Pull request merged**